### ISSUE TITLE
IPv6 addresses are not bracketed in In twisted.web.client._AgentBase, when setting HTTP host header

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8369]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)| @einarfd reported|
|-|-|
|Trac ID|trac#8369|
|Type|enhancement|
|Created|2016-05-30 08:43:48Z|
|Branch|https://github.com/twisted/twisted/tree/8369-einarfd-bracket_ipv6_host_header_8369|

When the host header is set in HTTP request in _AgentBase and subclasses, it is not bracketed if it is a IPv6 address.
This is a problem if the server tries to use the header to build a redirect location header. Different server handles this differently, but usually you end up with a mangled URL in the location header, that doesn't work. 

<details><summary>Searchable metadata</summary>

```
trac-id__8369 8369
type__enhancement enhancement
reporter__einarfd einarfd
priority__normal normal
milestone__None None
branch__8369_einarfd_bracket_ipv6_host_header_8369 8369-einarfd-bracket_ipv6_host_header_8369
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464597828724726 1464597828724726
changetime__1464856331133470 1464856331133470
version__None None
owner__einarfd einarfd

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)<a name="note_1"></a>|@einarfd commented|
|-|-|

Added pull request with fix on GitHub at https://github.com/twisted/twisted/pull/86

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @einarfd|
|-|-|

Changes looks good and the branch is in great shape.

Many thanks for the contribution.

I have updated the news fragment as outside of the Twisted core base people should not care about twisted.web.client._BaseAgent, but only about the public API.. so there is no point in talking about private API in the release notes.

Waiting for the tests to be green and will merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

einarfd the branch is all green.

Please confirm that you are happy with my changes and then I will merge the branch. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)<a name="note_5"></a>|@einarfd commented|
|-|-|

I'm happy with the changes, the text is better than the one I had.

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)|@einarfd set owner to @adiroiban|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set owner to @einarfd|
|-|-|

thanks . will merge

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 16c2d99c4121c0464644c2caeb5da544562c32e2
```
#!CommitTicketReference repository="" revision="16c2d99c4121c0464644c2caeb5da544562c32e2"
Merge 8369-einarfd-bracket_ipv6_host_header_8369: Fix ipv6 address literal brackets in t.w.client._AgentBase.

Author: einarfd
Reviewer: adiroiban
Fixes: #8369
```
